sentry-compose/src/androidMain/kotlin/io/sentry/compose/SentryNavigationIntegration.kt

```diff
@@ -73,3 +73,15 @@ public fun NavHostController.withSentryObservableEffect(
     }
     return this
 }
+
+/**
+ * A [DisposableEffect] that captures a [Breadcrumb] and starts an [ITransaction] and sends
+ * them to Sentry for every navigation event when being attached to the respective [NavHostController].
+ */
+@Composable
+public fun NavHostController.withSentryObservableEffect(): NavHostController {
```

Comment:
I'm still thinking if we should change it to `internal fun`, so those who manually instrument do not get confused between 2 methods? It should still work in bytecode, since it's effectively public, but for the user-written code the kotlin compiler will fail, can we try that?

Or in general having some way to hide it would be great, either via @Deprecated or @OptIn annotations, if the above approach doesn't work

Reply:
`internal` worked, looking good now!